src/extension.ts

```diff
@@ -33,7 +35,28 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
     traceVerbose(`Configuration: ${JSON.stringify(serverInfo)}`);
 
     const runServer = async () => {
-        client = await restartServer(serverId, serverName, outputChannel, client);
+        if (clientPromise) {
```

Comment:
clientPromise != null

Let's not depend on implicit conversion (it's safe here because it is a promise). We should add eslint or Rome

Reply:
Please add the appropriate tooling, i'm not up to date what the appropriate tooling in the js ecosystem is.

Reply:
i tried adding eslint and it didn't flag this, i think i got the wrong set of rules